src/platform_impl/windows/mod.rs

```diff
@@ -65,6 +68,51 @@ impl WindowId {
     }
 }
 
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct OsSpecificWindowEvent {
+    window: HWND,
+    msg: UINT,
+    wparam: WPARAM,
+    lparam: LPARAM,
+    retval: *mut Option<LRESULT>,
+}
+
+impl OsSpecificWindowEvent {
+    #[inline]
+    pub fn message(&self) -> u32 {
```

Comment:
What's the point of these setter-getters? Why not just make the member public? The only one which I think makes sense is `set_overriden_reply`, since it looks non-trivial, however, it's got no documentation that describes what it does!

Reply:
yes, maybe i should just expose them.

Reply:
Public fields does make more consistency with current `winit` apis, but after i start to make these member public, it occurs to me that then people will be able to set up random values for these fields, and use the unsafe fn `set_overriden_reply` to send them to the system API. (Which may crash the process.) While nobody will likely do this, does it make better sense to keep them read-only (only readable with getter) in the current way?

Reply:
Users fiddling with the `window`, `msg` `wparam` and `lparam` fields shouldn't do anything, right, since they are all copy?

Don't the values get implicitly cloned [here](https://github.com/rust-windowing/winit/pull/868/files#diff-d4d8c8dcf07874a35d08c6a70e0a6dafR1505), and hen don't the originals get sent [here](https://github.com/rust-windowing/winit/pull/868/files#diff-d4d8c8dcf07874a35d08c6a70e0a6dafR1505).

The only members which users fiddling with can do damage is `retval`, and I guess we can keep the getter/setter for that (and it is unsafe).

Reply:
I'd rather `retval` becomes `*mut LRESULT,`, and have it set by default to `commctrl::DefSubclassProc(self.window, self.msg, self.wparam, self.lparam)`. Then we can just have this:
```
    #[inline]
    pub unsafe fn set_reply(&self, val: isize) {
        *self.retval = val;
    }

    #[inline]
    pub unsafe fn set_overriden_reply(&self, f: impl FnOnce(isize) -> isize) {
        use winapi::um::commctrl;
        let retval = f(*self.retval);
        *self.retval = Some(retval);
    }
}
```

If users want to improperly fiddle with unsafe functions, that's their call.

Reply:
If we do that, we only have to return `retval` [here](https://github.com/rust-windowing/winit/pull/868/files#diff-d4d8c8dcf07874a35d08c6a70e0a6dafR1514)

Reply:
Hmmm, reading over the docs you wrote, I was under the assumption that `commctrl::DefSubclassProc` was some mapping function, and not a syscall.

We should keep `retval` as an `Option` and just let users fiddle with them all they want. We can then state in the docs for `set_overriden_reply` that fiddling with them incorrectly could result in UB.

Reply:
All `commctrl::DefSubclassProc()` does is "call the next handler, or default handler, if any", which at most times is what user want but sometimes is what user want to avoid(for example, the default handler might fill the window background with a color, causing flash). And when the user calls it, user may still decide to change the value it returns (this potential change is what i meant with the word mapping). Maybe the function names needs some bike shedding too?

Reply:
Changed these fields to public.